src/execution_plan/ops/op_update.c

```diff
@@ -167,4 +168,12 @@ static void UpdateFree(OpBase *ctx) {
 	}
 
 	raxStop(&op->it);
+
+	if(op->add_labels) {
+		RG_Matrix_free(&op->add_labels);
+	}
+
+	if(op->remove_labels) {
+		RG_Matrix_free(&op->remove_labels);
+	}
```

Comment:
I believe this logic needs to be also performed in `UpdateReset`
Generally speaking I see there's some diffs between `UpdateFree` and `UpdateReset` take this opportunity to review `UpdateReset` and make sure we haven't missed anything.